micrometer-core/src/test/java/io/micrometer/core/instrument/binder/jvm/JvmGcMetricsTest.java

```diff
@@ -103,6 +106,11 @@ void gcTimingIsCorrect() {
         checkCollectionTime(pauseTimeMs, concurrentTimeMs);
     }
 
+    @Test
+    @Issue("gh-2872")
+    void nonGenerationalSizeMetricsNotSetToZero() {
+    }
```

Comment:
Oops. I intended to write a unit test for this issue, and I thought it would be easier to do now that the listener is not an anonymous class. I struggled to figure out how to instantiate some of the management classes involved with the gc notifications to check the behavior fixed. Let me take another whack at it or remove this before merging.